Content.Client/GameObjects/Components/PDA/PDABoundUserInterface.cs

```diff
@@ -108,18 +125,96 @@ protected override void UpdateState(BoundUserInterfaceState state)
                             _menu.AddListingGui(item);
                         }
                     }
+
+                    var balance = _menu.CurrentLoggedInAccount.DataBalance;
+                    var weightedColor = GetWeightedColorString(balance);
+                    _menu.BalanceInfo.SetMarkup(Loc.GetString("TC Balance: [color={0}]{1}[/color]", weightedColor, balance));
+                    _menu.MasterTabContainer.SetTabVisible(1, msg.Account != null);
                     break;
                 }
-
             }
         }
 
+
         protected override void Dispose(bool disposing)
         {
             base.Dispose(disposing);
             _menu?.Dispose();
         }
 
+        /// <summary>
+        /// This is shitcode. It is, however, "PJB-approved shitcode".
+        /// </summary>
+        /// <param name="x"></param>
+        /// <returns></returns>
+        public static Color GetWeightedColor(int x)
+        {
+            var weightedColor = Color.Gray;
+            if (x <= 0)
+            {
+                return weightedColor;
+            }
+            if (x <= 5)
```

Comment:
Why not just make this `0 <= x && x <= 5` and take away the first conditional?

Reply:
that.. doesnt achieve the desired effect at all. resolved.